models/Comment.js

```diff
@@ -0,0 +1,34 @@
+const mongoose = require("mongoose");
+
+const CommentSchema = new mongoose.Schema({
+    post: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "Post"
+    },
+    message: {
+        type: String,
+        required: true
+    },
+    likes: {
+        type: Number,
+        required: true,
+    },
+    user: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+    },
+    userName: {
+        type: String,
+        required: true
+    },
+    createdAt: {
+        type: Date,
+        default: Date.now,
+    },
+    replies: {
+        type: [this],
```

Comment:
Here is where you can recursively add Comment replies as arrays of Comments (who will then each have their own Comment replies array), by referencing "this".